packages/koehnlab/io/basis_util.py

```diff
@@ -0,0 +1,85 @@
+import numpy as np
+
+def get_multispinmat_prod(s,spat_nums,coord):
+        """
+        Returns the spin matrix of all given Spins S in the productbasis of all spins and all spatial coordinates
+
+        Args:
+        --------------------------
+        s -- array of multiple spin states, sorted in descending order
+        spat_nums -- array of numbers of spatial states corresponding to the spin states
+        coord -- 0,1,2 for x,y,z
+
+        Returns:
+        --------------------------
+        S -- spin matrix in Productbasis of given spin and spatial states
+        """
+        multiplicities = []
+        for i in range(len(s)):
+                multiplicities.append(int(2*s[i]+1))
+        dim = int(np.sum(multiplicities*spat_nums))
+        S = np.zeros((dim,dim),dtype = complex)
```

Comment:
```suggestion
        Smat = np.zeros((dim,dim),dtype = complex)
```
(otherwise it'd be easy to get confused between a single quantum number `S`, the array of all spin quantum numbers `s` and the matrix representation of the spin operator)